src/modules/external/aha/components/AltAhaScore.tsx

```diff
@@ -0,0 +1,103 @@
+import { Stack, Typography } from '@mui/material';
+import { useCallback, useContext, useState } from 'react';
+import LabelWithContent from '@/components/elements/LabelWithContent';
+import LoadingButton from '@/components/elements/LoadingButton';
+import AssessmentContext from '@/modules/assessments/components/AssessmentContext';
+import ApolloErrorAlert from '@/modules/errors/components/ApolloErrorAlert';
+import ErrorAlert from '@/modules/errors/components/ErrorAlert';
+import { emptyErrorState, ErrorState, hasErrors } from '@/modules/errors/util';
+import { FormDefinitionHandlers } from '@/modules/form/hooks/useFormDefinitionHandlers';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import {
+  CalculateAltAhaScoreMutation,
+  useCalculateAltAhaScoreMutation,
+} from '@/types/gqlTypes';
+
+interface AltAhaScoreProps extends DynamicInputCommonProps {
+  value?: number | null;
+  onChange?: (value: number | null) => void;
+  handlers?: FormDefinitionHandlers;
+}
+
+// Shows a button to calculate the score based on current form values, and displays the result.
+const AltAhaScore = ({
+  value,
+  onChange,
+  label,
+  disabled = false,
+  handlers,
+}: AltAhaScoreProps) => {
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+
+  const [calculateAltAhaScore, { loading }] = useCalculateAltAhaScoreMutation({
+    onCompleted: (data: CalculateAltAhaScoreMutation) => {
+      const errors = data.calculateAltAhaScore?.errors || [];
+      if (errors.length > 0) {
+        setErrorState({ ...emptyErrorState, errors });
+        return;
+      }
+
+      setErrorState(emptyErrorState);
+
+      if (onChange && data.calculateAltAhaScore) {
+        onChange(data.calculateAltAhaScore.score || null);
+      }
+    },
+    onError: (apolloError: any) => {
+      setErrorState({ ...emptyErrorState, apolloError });
+    },
+  });
+
+  const { formDefinitionIdentifier, enrollmentId } =
+    useContext(AssessmentContext) || {};
+
+  const handleFetch = useCallback(() => {
+    if (!enrollmentId || !formDefinitionIdentifier) {
+      // In the preview environment, mock the response instead of calling the mutation
+      onChange?.(10);
+    } else {
+      calculateAltAhaScore({
+        variables: {
+          enrollmentId,
+          formDefinitionIdentifier: formDefinitionIdentifier,
+          valuesByLinkId: handlers?.getValuesForSubmit().valuesByLinkId || {},
+        },
+      });
+    }
+  }, [
+    enrollmentId,
+    onChange,
+    calculateAltAhaScore,
+    formDefinitionIdentifier,
+    handlers,
+  ]);
+
+  return (
+    <Stack direction='column' gap={1} alignItems='flex-start'>
+      {label}
+      {errorState && hasErrors(errorState) && (
+        <Stack gap={1} sx={{ mb: 1 }}>
+          <ApolloErrorAlert error={errorState.apolloError} />
```

Comment:
```suggestion
          <ApolloErrorAlert error={errorState.apolloError} inline />
```

This is appearing as a snackbar at the top of the screen but its out of view. (Ideally the snackbar should be sticky, but that wasn't easy for me to figure out quickly). Suggest presenting the error inline instead so the user can see it